src/background.js

```diff
@@ -823,20 +823,15 @@ function onMessageHandler(request, sender, callback) {
 	}
 	if (name === 'account.getUserSubscriptionData') {
 		account.getUserSubscriptionData()
-			.then((customer) => {
-				// TODO temporary fix to handle multiple subscriptions
-				let sub = customer.subscriptions;
-				if (!Array.isArray(sub)) {
-					sub = [sub];
-				}
+			.then((subscriptions) => {
+				// Return highest tier subscription from array
+				const premiumSubscription = subscriptions.find(subscription => subscription.productName.includes('Ghostery Premium'));
+				if (premiumSubscription) callback({ subscriptionData: premiumSubscription });
 
-				let subscriptionData = sub.find(subscription => subscription.productName.includes('Ghostery Premium'));
-				if (subscriptionData) {
-					callback({ subscriptionData });
-				}
+				const plusSubscription = subscriptions.find(subscription => subscription.productName.includes('Ghostery Plus'));
+				if (plusSubscription) callback({ subscriptionData: plusSubscription });
 
-				subscriptionData = sub.find(subscription => subscription.productName.includes('Ghostery Plus'));
-				callback({ subscriptionData });
+				callback({});
```

Comment:
Invoking the callback doesn't cause the function to return, so we need to add explicit return statements after the Premium and Plus callbacks. Looks good otherwise!

Reply:
Added explicit returns!